lib/file.js

```diff
@@ -176,24 +187,81 @@ var File = (function() {
     return path.resolve(path.join(this.baseDir, filePath));
   }
 
+
+  //
+  // Private Method
+  //
+
+
   // 必須の引数を確認する
   //
   // @param {string} arg 確認したい値
   // @throws {Error} argがundefinedかブランクだった場合、例外を発生する
   // @return {void}
-  p.checkRequiredArg = function(arg, paramName) {
-    if (arg == undefined || arg == '') { throw new Error(paramName + ' is required.'); }
-  };
+  function checkRequiredArg(arg, paramName) {
+    var d = $.Deferred();
+    if (_.isBlank(arg)) d.reject(new AppError(500, paramName + ' is required.'));
+    else d.resolve();
+    return d.promise();
+  }
 
   // filePathのPermissionを確認する。
   //
   // 読み込む許可があり、baseDir以下だったらtrue、そうじゃなければfalseを返す
   // @param {string} filePath チェックしたいファイルのパス
   // @return {boolean}
-  p.checkPermission = function(filePath) {
-    return filePath.lastIndexOf(this.baseDir, 0) === 0 ? true : false;
+  function checkPermission(baseDir, filePath) {
+    var d = $.Deferred();
+    if (_.startsWith(filePath, baseDir)) d.resolve(filePath);
```

Comment:
Expected '{' and instead saw 'd'.<br>'_' is not defined.<br>Too many errors. (79% scanned).